### PR TITLE
added Pending Async as status for transaction

### PR DESCRIPTION
**Pull request by brett-onions**
Updating backend to allow transaction status of "Pending Async" TB-162

### CODE REVIEW

File: src/middleware/messageStore.js [hunk 1]
```diff
@@ -93,9 +94,13 @@ export function storeTransaction(ctx, done) {
 
 export function storeResponse(ctx, done) {
   const headers = copyMapWithEscapedReservedCharacters(ctx.response.header)
+  
+  const status = ctx.matchingChannel.isAsynchronousProcess && ctx.response.status < 400 ? 202 : ctx.response.status;
+  // updates the status code that appears to for the client
```
Comment (bradsawadye) [inline src/middleware/messageStore.js:99]: typo

File: src/middleware/messageStore.js [hunk 2]
```diff
@@ -93,9 +94,13 @@ export function storeTransaction(ctx, done) {
...
+  ctx.response.status = status;
```
Comment (bradsawadye) [inline src/middleware/messageStore.js:100]: Remove statement terminator for consistency

File: src/middleware/messageStore.js [hunk 3]
```diff
@@ -279,6 +288,8 @@ export function setFinalStatus(ctx, callback) {
 
       logger.info(`Final status for transaction ${tx._id} : ${tx.status}`)
       update.status = tx.status
+      
+
```
Comment (bradsawadye) [inline src/middleware/messageStore.js:292]: Remove empty lines

File: src/model/channels.js [hunk 4]
```diff
@@ -112,6 +112,11 @@ const ChannelDef = {
     type: String,
     required: true
   },
+  isAsynchronousProcess:{
+    type: Boolean,
+    required: false,
```
Comment (bradsawadye) [inline src/model/channels.js:117]: default for required is false. We can remove this line

**Comment by bradsawadye** [comment]
@brett-onions looks good. Please add some tests

**Review by Lumiere-Lodi — APPROVED** [review]
Looks good now with the tests.

File: src/middleware/messageStore.js [hunk 5]
```diff
@@ -253,7 +257,11 @@ export function setFinalStatus(ctx, callback) {
           ctx.response.status <= 299 &&
           routeSuccess
         ) {
-          tx.status = transactionStatus.SUCCESSFUL
+          if(ctx?.matchingChannel?.isAsynchronousProcess || !!ctx?.authorisedChannel?.isAsynchronousProcess){
+            tx.status = transactionStatus.PENDING_ASYNC
```
Comment (nour-borgi) [inline src/middleware/messageStore.js:261]:
```suggestion
          if(isAsynchronousProcess){
            tx.status = transactionStatus.PENDING_ASYNC
```

And please don't forget to run `npm run lint`  and `npm run lint:fix` to lint these changes

**Review by nour-borgi — CHANGES_REQUESTED** [review]

**Review by arran-standish — DISMISSED** [review]
looks good

File: src/middleware/messageStore.js [hunk 6]
```diff
@@ -253,7 +264,14 @@ export function setFinalStatus(ctx, callback) {
           ctx.response.status <= 299 &&
           routeSuccess
         ) {
-          tx.status = transactionStatus.SUCCESSFUL
+          if (
+            ctx?.matchingChannel?.isAsynchronousProcess ||
+            !!ctx?.authorisedChannel?.isAsynchronousProcess
+          ) {
```
Comment (nour-borgi) [inline src/middleware/messageStore.js:270]:
```suggestion
          if (isAsynchronousProcess) {
```

**Review by nour-borgi — APPROVED** [review]
LGTM

**Review by nour-borgi — APPROVED** [review]
LGTM

**Comment by MatthewErispe** [comment]
this is pending bug fix https://jembiprojects.jira.com/browse/TB-202

**Review by MatthewErispe — DISMISSED** [review]
@brett-onions happy with the bug fix change

**Review by nour-borgi — APPROVED** [review]

**Review by MatthewErispe — APPROVED** [review]

**Pull request merged**